cluster-provision/gocli/cmd/PROVISION_MANAGER.md

```diff
@@ -0,0 +1,49 @@
+# Provision Manager
+
+## Purpose:
+Detect which files are changed versus the latest `KUBEVIRTCI_TAG`,
+and build only the providers which should be updated.
+The other providers will be reused by retagging their latest version.
+
+## Motivation:
+1. Freeze a state unless a change happens.
+2. Faster provision job.
+3. Reduce collision domain, a problem on one provider won't block updating the others.
+
+## Related Files:
+1. `hack/pman/rules.txt` - the rules that determine which folder/file affects which provider.
+2. `hack/pman/force` - changing this file allows to enforce rebuild of all vm based providers.
```

Comment:
Why not adding a flag to the pman command?

Reply:
You need a way to change a file in your commit, which trigger the pre submit / post submit to rebuild all the targets.
So you can use a flag only manually, but not on CI, unless you have another lane which you will run manually, but it is prone to forget.

You also need that it will affect just the current PR, so the logic of pman itself is used here - if the file isn't changed it won't redo it, so it will affect only one time.